gobblin-hive-registration/src/main/java/gobblin/hive/HiveMetaStoreClientFactory.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.hive;
+
+import org.apache.commons.pool.BasePoolableObjectFactory;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+
+/**
+ * An implementation of {@link BasePoolableObjectFactory} for {@link HiveMetaStoreClient}.
+ *
+ * @author ziliu
+ */
+public class HiveMetaStoreClientFactory extends BasePoolableObjectFactory<HiveMetaStoreClient> {
+
+  @Override
+  public HiveMetaStoreClient makeObject() throws Exception {
```

Comment:
This should probably return `IMetaStoreClient` instead as according to the java docs that seems to be the public API:

https://hive.apache.org/javadocs/r1.0.1/api/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.html
https://hive.apache.org/javadocs/r1.0.1/api/org/apache/hadoop/hive/metastore/IMetaStoreClient.html